debvader/deblender.py

```diff
@@ -179,95 +161,62 @@ def get_predicted_field(self, res_deblend=None):
-                    )
-                    output_images_mean_padded[
-                        pad_start:pad_end, pad_start:pad_end, :
-                    ] = output_images_mean
-
-                    # Create the corresponding standard deviation image (aleatoric uncertainty).
-                    output_images_stddev_padded = np.zeros(
-                        (self.field_size, self.field_size, self.nb_of_bands)
-                    )
-                    output_images_stddev_padded[
-                        pad_start:pad_end, pad_start:pad_end, :
-                    ] = output_images_stddev
-
-                    x_pos = (
-                        isolated_galaxy_row["galaxy_distances_to_center_x"]
-                        + isolated_galaxy_row["shifts"][0]
-                    )
-                    y_pos = (
-                        isolated_galaxy_row["galaxy_distances_to_center_y"]
-                        + isolated_galaxy_row["shifts"][1]
-                    )
+                    x_shift = isolated_galaxy_row["shifts"][0]
+                    y_shift = isolated_galaxy_row["shifts"][1]
 
                     for band in range(
                         self.nb_of_bands
                     ):  # scipy.ndimage.shift is slow so looping over bands is faster
 
-                        denoised_field[:, :, band] += scipy.ndimage.shift(
-                            output_images_mean_padded[:, :, band], shift=(x_pos, y_pos)
+                        output_image_mean[:, :, band] += scipy.ndimage.shift(
+                            output_image_mean[:, :, band], shift=(x_shift, y_shift)
                         )
-                        denoised_field_std[:, :, band] += scipy.ndimage.shift(
-                            output_images_stddev_padded[:, :, band],
-                            shift=(x_pos, y_pos),
+                        output_image_stddev[:, :, band] += scipy.ndimage.shift(
+                            output_image_stddev[:, :, band],
+                            shift=(x_shift, y_shift),
                         )
 
-                        output_images_epistemic_padded = np.zeros(
-                            (self.field_size, self.field_size, self.nb_of_bands)
-                        )
                         if self.epistemic_uncertainty_estimation:
                             # Create the corresponding epistemic uncertainty image (aleatoric uncertainty).
-                            output_images_epistemic_padded[
-                                pad_start:pad_end, pad_start:pad_end, :
-                            ] = np.array(epistemic_uncertainty)
-                            denoised_field_epistemic[:, :, band] += scipy.ndimage.shift(
-                                output_images_epistemic_padded[:, :, band],
-                                shift=(x_pos, y_pos),
+                            epistemic_uncertainty = np.array(epistemic_uncertainty)
```

Comment:
what is this line for ?

Reply:
Good question!
It is a list containing a numpy array (no idea why), I'll fix this!